### PR TITLE
Fixing taskgraph issue on mac for n_workers > 0

### LINKED ISSUE #420: Models hang on macOS with n_workers>0
Seen on a few models but likely not a model-specific issue. May be related to the update to taskgraph 0.10
```
2020-12-11 13:39:47,889 Task.__del__(418) ERROR Expected an empty logging queue, but if the TaskGraph were being terminated it's possible this object is corrupt and we'd get a different kind of exception like EOF. In any case we should always stop trying to drain the queue in the case of an Exception.
Traceback (most recent call last):
  File "site-packages\taskgraph\Task.py", line 413, in __del__
  File "multiprocessing\queues.py", line 126, in get_nowait
  File "multiprocessing\queues.py", line 107, in get
_queue.Empty
2020-12-11 13:39:47,890 Task.__del__(440) DEBUG joining _monitor_thread.
2020-12-11 13:39:47,890 Task.__del__(464) ERROR exception occurred during __del__
Traceback (most recent call last):
  File "site-packages\taskgraph\Task.py", line 444, in __del__
  File "threading.py", line 1041, in join
RuntimeError: cannot join current thread
```

### PR DESCRIPTION
**Pull request by phargogh**
This PR addresses #420 by bumping the taskgraph requirement.

@emlys could you check that the mac binaries run as expected on your computer as well?

# Description
Fixes #420 

# Checklist
- [x] Updated HISTORY.rst (if these changes are user-facing)

- ~~[ ] Updated the user's guide (if needed)~~

### CODE REVIEW

**Review by emlys — APPROVED** [review]
@phargogh thank you for figuring this one out! The binaries work for me and DelineateIt isn't hanging any more. I'm still kind of concerned that the taskgraph update didn't resolve the bug natcap/taskgraph#43 for me. Is there something about the binary build environment that we don't understand, that's necessary for this to work? Approving, but I'd like to keep investigating this.

**Comment by phargogh** [comment]
> Is there something about the binary build environment that we don't understand, that's necessary for this to work? Approving, but I'd like to keep investigating this.

@emlys please do!

File: doc/api-docs/scripting.rst [hunk 1]
```diff
@@ -260,6 +263,44 @@ for many more complex logging features.  For more
...
+  InVEST)
+* ``0`` representing threaded task management
+* Any other positive integer represents the number of processes that will be
+  created to handle tasks.  ``2*multiprocessing.cpu_count()`` is usually a good
+  number.
+
+.. warning::
+   If you use this feature, you **must** wrap your script in a
+   ``if __name__ == '__main__':`` condition.  Failure to do so will result
+   in a fork bomb (https://en.wikipedia.org/wiki/Fork_bomb).
```
Comment (emlys) [inline doc/api-docs/scripting.rst:287]: Hah, I didn't know there was a term for it!

**Review by emlys — APPROVED** [review]

File: doc/api-docs/scripting.rst (hunk 1 above)
Comment (dcdenu4) [reply doc/api-docs/scripting.rst:287]: I didn't know about fork bombs either!

**Pull request merged**